### ISSUE TITLE
Rule proposal: generic-spacing

### ISSUE THREAD

**Issue by jonnytest1** [typescript-eslint/typescript-eslint#4433]
### Description

```
class Test<   Abc  >{...} 

class Abc{
    constructor(private test:Test<  Def    >){
    }
}
```

and other similar occurences of generics should be enforcable in wether they have spacing after the '<' and before the '>' 

as a default i recommend 'never'

### Fail

```ts
class Test<   Abc  >{...} 

class Abc{
    constructor(private test:Test<  Def    >){
    }
}
```

### Pass

```ts
class Test<Abc>{...} 

class Abc{
    constructor(private test:Test<Def>){
    }
}
```

**Comment by bradzacher** [comment]
Thanks for the suggestion!

Unfortunately we aren't accepting new stylistic rules into this project.
Stylistic rules are opinionated, difficult to implement, cumbersome to maintain, and usually require many options to satisfy everyone's opinions.

Our recommendation is to use a tool like prettier and do-away with the need for stylistic lint rules altogether.

You're welcome to leverage our tooling to build and publish such a lint rule externally!
